src/Messages/OWSMessageSender.m

```diff
@@ -554,6 +560,9 @@ - (void)handleMessageSentLocally:(TSOutgoingMessage *)message
 {
     [self saveMessage:message withState:TSOutgoingMessageStateSent];
     if (message.shouldSyncTranscript) {
+        // TODO: I suspect we shouldn't optimistically set hasSyncedTranscript.
+        //       We could set this in a success handler for [sendSyncTranscriptForMessage:].
```

Comment:
I agree it's weird to optimistically set it like this. But make sure than any changes don't result in it being sent once per group recipient.

E.g. if I'm in a group with Alice, Bob, and Georgia, I should still send only one sync message.